kmk/modules/encoder.py

```diff
@@ -58,19 +58,21 @@ def update_state(self):
                     self._direction = -1
 
             # when the encoder settles on a position (every 2 steps)
-            if new_state == (True, True):
-                if self._movement > 2:
-                    # 1 full step is 4 movements, however, when rotated quickly,
-                    # some steps may be missed. This makes it behaves more
-                    # naturally
-                    real_movement = round(self._movement / 4)
+            if new_state[0] == new_state[1]:
+                # when the encoder made a full loop according to its resolution
+                if self._movement >= self.resolution - 1:
+                    # 1 full step is 4 movements (2 for high-resolution encoder),
+                    # however, when rotated quickly, some steps may be missed.
+                    # This makes it behave more naturally
+                    real_movement = round(self._movement / self.resolution)
```

Comment:
```suggestion
                    real_movement = self._movement // self.resolution
```
prefer integer division, needs test on hardware